client/landing/stepper/declarative-flow/internals/steps-repository/store-profiler/index.tsx

```diff
@@ -30,7 +31,8 @@ const StoreProfiler: Step = function StoreProfiler( { navigation } ) {
 		setStoreLocationCountryCode: saveCountryCodeToStore,
 	} = useDispatch( ONBOARD_STORE );
 
-	const verticals = useSiteVerticalsFeatured();
+	const userSettings = useUserSettingsQuery();
+	const verticals = useSiteVerticalsFeatured( { locale: userSettings?.data?.language || 'en' } );
```

Comment:
The `useUserSettingsQuery()` relies on a query back to the server, so it feels hugely inefficient, as we are already relying on `useTranslate()/translate()` to perform translations. I think it would be better to rely on `getLocaleSlug()` from `i18n-calypso` to return the current locale.
```suggestion
	const userLocale = getLocale() || 'en';
	const verticals = useSiteVerticalsFeatured( { locale: userLocale } );
```

_Sidebar:_ I also checked and the WordPress core code also has a `useLocale()` hook, but that uses the `__()` translation function. We don't want to mix the two in this code.